pkg/kev/config/extensions.go

```diff
@@ -0,0 +1,143 @@
+package config
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/imdario/mergo"
+	"gopkg.in/yaml.v3"
+)
+
+type ExtensionRoot struct {
+	K8S K8SServiceConfig `yaml:"x-k8s"`
+}
+
+type K8SServiceConfig struct {
+	Enabled   bool      `yaml:"enabled"`
+	Workload  Workload  `yaml:"workload" validate:"required"`
+	Component Component `yaml:"component"`
+}
+
+func DefaultK8SServiceConfig() K8SServiceConfig {
+	return K8SServiceConfig{
+		Enabled: DefaultServiceEnabled,
+		Workload: Workload{
+			Type:           DefaultWorkload,
+			LivenessProbe:  DefaultLivenessProbe(),
+			ReadinessProbe: DefaultReadinessProbe(),
+		},
+	}
+}
+
+func K8SServiceCfgFromMap(m map[string]interface{}) (K8SServiceConfig, error) {
+	var extensions ExtensionRoot
+
+	var buf bytes.Buffer
+	if err := yaml.NewEncoder(&buf).Encode(m); err != nil {
+		return K8SServiceConfig{}, err
+	}
+
+	if err := yaml.NewDecoder(&buf).Decode(&extensions); err != nil {
+		return K8SServiceConfig{}, err
+	}
+
+	err := validator.New().Struct(extensions.K8S.Workload)
+	if err != nil {
+		validationErrors := err.(validator.ValidationErrors)
+		for _, e := range validationErrors {
+			if e.Tag() == "required" {
+				return K8SServiceConfig{}, fmt.Errorf("%s is required", e.StructNamespace())
+			}
+		}
+
+		return K8SServiceConfig{}, errors.New(validationErrors[0].Error())
+	}
+
+	k8s := DefaultK8SServiceConfig()
+
+	if err := mergo.Merge(&extensions.K8S, k8s); err != nil {
+		return K8SServiceConfig{}, err
+	}
+
+	return extensions.K8S, nil
+}
+
+// TODO: hyphenated words suck, should we do camelCase?
+type Workload struct {
+	Type           string         `yaml:"type"`
+	Replicas       int            `yaml:"replicas" validate:"required"`
+	LivenessProbe  LivenessProbe  `yaml:"liveness-probe"`
+	ReadinessProbe ReadinessProbe `yaml:"readiness-probe"`
+}
+
+type LivenessProbe struct {
+	// TODO: find a decent way of using ProbeType here that validates the content of the string
+	Type        string `yaml:"type" validate:"required"`
+	ProbeConfig `yaml:",inline"`
+}
+
+func DefaultLivenessProbe() LivenessProbe {
+	id, _ := time.ParseDuration(DefaultProbeInitialDelay)
+	p, _ := time.ParseDuration(DefaultProbeInterval)
+
+	return LivenessProbe{
+		Type: ProbeTypeExec.String(),
+		ProbeConfig: ProbeConfig{
```

Comment:
`ProbeConfig` missing timeout option - also applicable to the ReadinessProbe below. 

It'd be great to have this (liveness/readiness probe) scoped portion of extension based config work as complete as we possibly can to cover the current state of play with the probes. This way we'll spend less time context switching between different parts in the future PRs. 

Transition of remaining configuration will be then scoped around their respective options in the follow up PR which will be a nice way of braking that work down in stages.